components/playlist/browser/playlist_download_request_manager.cc

```diff
@@ -329,43 +245,120 @@ void PlaylistDownloadRequestManager::ProcessFoundMedia(
       item->thumbnail_source = GURL(*thumbnail);
       item->thumbnail_path = GURL(*thumbnail);
     }
-    item->media_source = GURL(*src);
-    item->media_path = GURL(*src);
-    if (author)
+    if (author) {
       item->author = *author;
-
+    }
     items.push_back(std::move(item));
   }
 
-  DVLOG(2) << __func__
-           << " Media detection result size: " << value.GetList().size() << " "
-           << items.size();
-  std::move(callback).Run(std::move(items));
+  DVLOG(2) << __func__ << " Media detection result size: " << items.size();
+
+  if (contents == background_contents() && items.size()) {
+    CHECK(!callback_for_current_request_.is_null()) << " callback already ran";
+    auto callback = std::move(callback_for_current_request_);
+
+    DCHECK_GT(in_progress_urls_count_, 0);
+    in_progress_urls_count_--;
+
+    std::vector<mojom::PlaylistItemPtr> cloned_items;
+    base::ranges::transform(items, std::back_inserter(cloned_items),
+                            &mojom::PlaylistItemPtr::Clone);
+    std::move(callback).Run(std::move(cloned_items));
+
+    web_contents_.reset();
+  }
+
+  return items;
 }
 
+bool PlaylistDownloadRequestManager::CanCacheMedia(
+    const mojom::PlaylistItemPtr& item) const {
+  GURL media_url(item->media_source);
+  if (media_url.SchemeIs(url::kHttpsScheme)) {
+    return true;
+  }
+
+  if (media_url.SchemeIsBlob()) {
+    if (item->is_blob_from_media_source) {
+      // At this moment, we have a few sites that we can get media files with
+      // hacks.
+      return media_detector_component_manager_->ShouldHideMediaSrcAPI(
+                 media_url) ||
+             media_detector_component_manager_->ShouldUseFakeUA(media_url);
+    }
+
+    // blob: which is not Media Source
+    // TODO(sko) Test and allow this case referring to
+    // https://github.com/brave/brave-core/pull/17246
+    return false;
+  }
+
+  return false;
+}
```

Comment:
Similarly, here I think we should have something like:
```cpp
bool PlaylistDownloadRequestManager::CanCacheMedia(
    const mojom::PlaylistItemPtr& item) const {
  return item->media_source.SchemeIs(url::kHttpsScheme) ||
         item->is_blob_from_media_source;
}
```
1. if scheme is HTTPS, this returns `true` :white_check_mark:
2. if scheme is not HTTPS, nor `blob`, this returns `false` :white_check_mark:
3. if scheme is `blob`, but not a `MediaSource`-backed one, this returns `false` :white_check_mark:
4. if scheme is `blob`, and is a `MediaSource`-backed one, this returns `true` :white_check_mark:

Which is exactly what we support: HTTPS URLs, and `blob`s that are `MediaSource`-based.
And again, without taking into consideration whether `media_detector_component_manager_->ShouldUseFakeUA(item->media_source)` is `true`, or not — `PlaylistService` will include `should_force_fake_ua` in the `PlaylistDownloadRequestManager::Request` where appropriate.